Source/Core/VideoCommon/OnScreenDisplay.cpp

```diff
@@ -56,7 +59,7 @@ void DrawMessages()
 
 		alpha <<= 24;
 
-		g_renderer->RenderText(it->str, left, top, 0xffff30 | alpha);
+		g_renderer->RenderText(it->m_str, left, top, (it->m_rgb & 0xFFFFFF) | ((it->m_rgb >> 24) * alpha));
```

Comment:
float alpha = max(1.0, min(0.0, time_left / 1024.0))
u32 color = it->m_rgb & 0xFFFFFF;
color |= (u32)((it->m_rgb >> 24) * alpha) << 24;
g_renderer->RenderText(it->m_str, left, top, color);